usi/src/main/java/daq/usi/ovs/OpenVSwitch.java

```diff
@@ -11,61 +11,64 @@
 import java.io.BufferedReader;
 import java.io.FileNotFoundException;
 import java.io.FileReader;
+import java.io.IOException;
 import java.net.URL;
+import java.util.concurrent.TimeUnit;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class OpenVSwitch implements SwitchController {
 
   private static final String OVS_OUTPUT_FILE = "ovs_output.txt";
 
-  protected String getInterfaceByPort(int devicePort) throws FileNotFoundException {
+  protected String getInterfaceByPort(int devicePort) throws IOException {
     URL file = OpenVSwitch.class.getClassLoader().getResource(OVS_OUTPUT_FILE);
     if (file == null) {
       throw new FileNotFoundException(OVS_OUTPUT_FILE + " is not found!");
     }
     FileReader reader = new FileReader(file.getFile());
-    BufferedReader bufferedReader = new BufferedReader(reader);
-    Pattern pattern = Pattern.compile("(^\\s*" + devicePort + ")(\\((.+)\\))(:.*)", 'g');
-    String interfaceLine = bufferedReader.lines().filter(line -> {
-      Matcher m = pattern.matcher(line);
-      return m.find();
-    }).findFirst().get();
-
-    Matcher m = pattern.matcher(interfaceLine);
-    m.matches();
-    return m.group(3);
+    try (BufferedReader bufferedReader = new BufferedReader(reader)) {
+      Pattern pattern = Pattern.compile("(^\\s*" + devicePort + ")(\\((.+)\\))(:.*)", 'g');
+      String interfaceLine = bufferedReader.lines().filter(line -> {
+        Matcher m = pattern.matcher(line);
+        return m.find();
+      }).findFirst().get();
+      bufferedReader.close();
```

Comment:
close is handled implicitly by the try clause, so you shouldn't do it explicitly here